compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -546,14 +591,16 @@ public void visit(BLangSimpleVariable varNode) {
             String closureName = generateName(varNode.symbol.name.value, env.node);
             varNode.pos = null;
             varNode.expr.pos = null;
-            generateClosureForDefaultValues(closureName, varNode.name.value, varNode);
+            generateClosureForDefaultValues(closureName, varNode.name.value, varNode.expr, varNode.getBType(),
+                                            env.node.getBType().tsymbol);
             result = varNode;
             return;
         }
 
-        if (varNode.symbol != null && Symbols.isFlagOn(varNode.symbol.flags, Flags.DEFAULTABLE_PARAM)) {
+        if (Symbols.isFlagOn(varNode.symbol.flags, Flags.DEFAULTABLE_PARAM)) {
             String closureName = generateName(varNode.symbol.name.value, env.node);
-            generateClosureForDefaultValues(closureName, varNode.name.value, varNode);
+            generateClosureForDefaultValues(closureName, varNode.name.value, varNode.expr, varNode.getBType(),
+                                            env.node.getBType().tsymbol);
```

Comment:
I would introduce an overloaded method to avoid repeating the calls to generateClosureForDefaultValues with this expanded param list.